fs/inode/fs_files.c

```diff
@@ -298,7 +301,10 @@ int files_duplist(FAR struct filelist *plist, FAR struct filelist *clist)
 #endif
 
           filep = &plist->fl_files[i][j];
-          if (filep->f_inode == NULL || (filep->f_oflags & O_CLOEXEC) != 0)
+          DEBUGASSERT(filep);
+
+          if (filep && (filep->f_inode == NULL ||
+                       (filep->f_oflags & O_CLOEXEC) != 0))
```

Comment:
I do not understand this. Why do we need `DEBUGASSERT(filep);` and `if (filep`? I would expect code to be
```suggestion
          DEBUGASSERT(filep);

          if (filep->f_inode == NULL ||
              (filep->f_oflags & O_CLOEXEC) != 0)
```

Reply:
Actually the original fix as only that "if (filep && (...))", I added the DEBUGASSET later because it will avoid some silent issue that could happen in some cases.

Reply:
But in such case if `filep` is checked unconditionally shouldn't we move the assert before `continue`, so that will eliminate unnecessary duplicated NULL check?